include/rfl/internal/num_fields.hpp

```diff
@@ -97,20 +108,68 @@ struct CountFieldsHelper {
     }
   }
 
+  template <std::size_t size = 0>
+  static consteval std::size_t get_the_sole_nested_array_size() {
+    static_assert(size <= sizeof(T));
+    if constexpr (constructible_with_nested<0, size, 0>() &&
+                  !constructible_with_nested<0, size + 1, 0>()) {
+      return size;
+    } else {
+      return get_the_sole_nested_array_size<size + 1>();
+    }
+  }
+
+  template <std::size_t n, std::size_t total_arg_num>
+  static consteval bool has_n_base_param() {
+    if constexpr (n > total_arg_num) {
+      return false;
+    } else {
+      auto left = std::make_index_sequence<n>();
+      auto right = std::make_index_sequence<total_arg_num - n>();
+      return []<std::size_t... l, std::size_t... r>(std::index_sequence<l...>,
+                                                    std::index_sequence<r...>) {
+        return requires { T{any_base<T>(l)..., any(r)...}; };
+      }(left, right);
+    }
+  }
+
+  template <std::size_t total_arg_num, std::size_t index>
+  static consteval std::size_t base_param_num() {
+    if constexpr (!has_n_base_param<index + 1, total_arg_num>()) {
+      return index;
+    } else {
+      return base_param_num<total_arg_num, index + 1>();
+    }
+  }
+
   template <std::size_t index, std::size_t max>
-  static consteval std::size_t count_fields_impl() {
+  static consteval std::size_t constructable_no_brace_elision() {
     static_assert(index <= max);
     if constexpr (index == max) {
       return 0;
     } else {
       return 1 +
-             count_fields_impl<
+             constructable_no_brace_elision<
                  index + get_nested_array_size<index, max - index, 0>(), max>();
     }
   }
 
   static consteval std::size_t count_fields() {
-    return count_fields_impl<0, count_max_fields()>();
+    constexpr std::size_t max_fields = count_max_fields();
+    constexpr std::size_t total_args =
+        constructable_no_brace_elision<0, max_fields>();
+    constexpr std::size_t base_args = base_param_num<total_args, 0>();
+    if constexpr (total_args == 0 && base_args == 0) {
+      // Empty struct
+      return 0;
+    } else if constexpr (base_args == total_args) {
+      // Special case when the derived class is empty.
+      // In such cases the filed number is the fields in base class.
+      // Note that there should be only one base class in this case.
```

Comment:
This is only true with this implementation, but not necessarily. 

You can check which base class is 'empty' and which is not with another ubiq: 
```cpp

template <class Derived>
struct any_empty_base {
  any_empty_base(std::size_t);
  template <class Base>
    requires(
        std::is_empty_v<std::remove_cvref_t<Base>> && // new check
        std::is_base_of_v<std::remove_cvref_t<Base>,
                          std::remove_cv_t<Derived>> &&
        !std::is_same_v<std::remove_cvref_t<Base>, std::remove_cv_t<Derived>>)
  constexpr operator Base&() const noexcept;
};
```

`T(any_empty_base(Bx...), any_base(N),  any_empty_base(Ax...))`. 
If no one is creatable (N <- [1..base_args]) then it is not a structure bindable class (static_assert).